### PR TITLE
immer settings

### PR DESCRIPTION
**Pull request by chrisnojima**
Immerize settings. Likely easiest to compare the reducer side by side

### CODE REVIEW

File: shared/settings/invites/index.desktop.tsx [hunk 1]
```diff
@@ -24,6 +24,11 @@ class Invites extends Component<Props, State> {
       showMessageField: props.showMessageField,
     }
   }
+
+  componentDidMount() {
+    this.props.onRefresh()
```
Comment (jakob223) [inline shared/settings/invites/index.desktop.tsx]: why add this?

File: shared/settings/notifications/container.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const ReloadableNotifications = (props: Props & ExtraProps) => {
   )
 }
 
-export default Container.connect(
+export default Container.connectDEBUG(
```
Comment (jakob223) [inline shared/settings/notifications/container.tsx]: 🔥

File: shared/signup/email/container.tsx [hunk 3]
```diff
@@ -39,7 +39,7 @@ const WatchForSuccess = (props: WatcherProps) => {
   )
 }
 
-const ConnectedEnterEmail = Container.connect(
+const ConnectedEnterEmail = Container.connectDEBUG(
```
Comment (jakob223) [inline shared/signup/email/container.tsx]: 🔥

File: shared/reducers/settings.tsx [hunk 4]
```diff
@@ -14,275 +13,282 @@ type Actions =
...
+    const {list} = action.payload.params
+    draftState.email.emails = new Map(
+      (list || []).map(row => [row.email, {...Constants.makeEmailRow(), ...row}])
+    )
+  },
+  [SettingsGen.emailVerified]: (draftState, action) => {
+    const {email} = action.payload
+    const map = draftState.email.emails
+    const old = map?.get(email)
+    if (old) {
```
Comment (jakob223) [inline shared/reducers/settings.tsx]: else log?

File: shared/settings/invites/index.desktop.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/settings/invites/index.desktop.tsx]: the container used to do this using recompose: https://github.com/keybase/client/pull/20953/files#diff-c14348ae521b4dd883e4c51536e266d8L47

**Comment by chrisnojima** [comment]
fixed issues. chat headers work w/ rebuilt daemon. invites work fine for me

**Comment by chrisnojima** [comment]
@jakob223 can you look at the recent commits, merges your new changes back in

File: shared/settings/invites/container.tsx [hunk 5]
```diff
@@ -8,7 +8,7 @@ import * as Container from '../../util/container'
 
 type OwnProps = {}
 
-export default Container.connect(
+export default Container.connectDEBUG(
```
Comment (jakob223) [inline shared/settings/invites/container.tsx]: 🔥

**Pull request merged**